docs/source/_k2/ragged/tensor.py

```diff
@@ -0,0 +1,564 @@
+from typing import Optional, Tuple, Union, overload
+
+
+import torch
+
+
+class RaggedShape:
+    pass
+
+
+class Tensor(object):
```

Comment:
I'm not sure that it is is a good idea to call this simply Tensor, IMO there is too much pontential for confusion here(?)  RaggedTensor might be a possibility?   Or just Ragged, maybe?

Reply:
The `Tensor` is put inside the module `k2.ragged`; its fully qualified name is `k2.ragged.Tensor`.

If it is called `RaggedTensor`, shall we put it in `k2`, or `k2.ragged`? That is, shall we call it
`k2.RaggedTensor` or `k2.ragged.RaggedTensor`?

By the way, this file is going to be deleted as the documentation is now contained in the C++ header files.
This pull request is based on #812 .

If you like the idea (i.e., moving all possible parts from Python to C++), I will close this pullrequest and
add commits from the pull-request to #812

Reply:
I think we could maybe keep it as k2.ragged.RaggedTensor but also import it into k2.  
I still have to think about this.  I'm impressed that you figured out how to do it, and we probably need to do this at
some point, but for me the main question in my mind is what is going to happen to class k2.Fsa.  That has a bunch of
logic, e.g. about attributes, that I wonder how it will be transferred to C++; and once it is C++, who will be able to
maintain it!